osu.Game.Rulesets.Taiko/Mods/TaikoModAlternate.cs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Scoring;
+using osu.Game.Rulesets.Taiko.Objects;
+
+namespace osu.Game.Rulesets.Taiko.Mods
+{
+    public class TaikoModAlternate : ModAlternate<TaikoHitObject, TaikoAction>
+    {
+        private TaikoAction? lastRimAction;
+        private TaikoAction? lastCentreAction;
+
+        protected override void OnBreakEnd()
+        {
+            resetStates();
+        }
+
+        protected override void OnInterceptorLoadComplete()
+        {
+            Drawable interceptor = Interceptor as Drawable;
+
+            using (Interceptor.BeginAbsoluteSequence(0))
+            {
+                foreach (TaikoHitObject hitObject in HitObjects)
+                {
+                    if (!hitObject.IsStrong) continue;
+
+                    var window = hitObject.HitWindows.WindowFor(HitResult.Miss);
+                    interceptor.Delay(hitObject.StartTime - window).Schedule(() => resetStates());
+                    interceptor.Delay(hitObject.StartTime + window).Schedule(() => resetStates());
+
+                    foreach (TaikoHitObject h in hitObject.NestedHitObjects)
+                    {
+                        interceptor.Delay(h.StartTime - 50).Schedule(() => resetStates());
```

Comment:
What is `50` and why is it hard-coded?

Reply:
The hit window is based off of here.
https://github.com/ppy/osu/blob/fb5a54d2422171d8f2cb40ee1cb4b9b96dabc4f7/osu.Game.Rulesets.Taiko/Objects/DrumRoll.cs#L53-L57

Reply:
That value is not always valid, though; it's merely a default. Check other usages in `DrumRoll` (it is mutated based on timing points and tick rate).